effectful/handlers/pyro.py

```diff
@@ -252,190 +265,264 @@ def __repr__(self):
         return f"Naming({self.name_to_dim})"
 
 
-class PositionalDistribution(pyro.distributions.torch_distribution.TorchDistribution):
-    """A distribution wrapper that lazily converts indexed dimensions to
-    positional.
-
-    """
-
-    indices: Mapping[Operation[[], int], int]
-
-    def __init__(
-        self, base_dist: pyro.distributions.torch_distribution.TorchDistribution
-    ):
-        self.base_dist = base_dist
-        self.indices = sizesof(base_dist)
+@defop
+def named_distribution(
+    d: Annotated[TorchDistribution, Scoped[A]],
```

Comment:
Per my comment below on reenabling `Scoped._check_no_boundvars_in_result`, I think `named_distribution` is the only operation whose signature violates that condition, so changing the annotation on `d` here to
```py
d: Annotated[TorchDistribution, Scoped[A | B]]
```
should be sufficient to safely reenable the check without being an egregiously incorrect description of `named_distribution`.